### ISSUE TITLE
Add OptionalBooleanParameter AllowedMethods configuration

### ISSUE THREAD

**Issue by bkuhlmann** [rubocop-hq/rubocop#8489]
## Describe the solution you'd like

When using `Style/OptionalBooleanParameter`, it would be nice to define a set of allowed methods as used via [Style/NestedParenthesizedCalls](https://docs.rubocop.org/rubocop/cops_style.html#stylenestedparenthesizedcalls), for example. For instance, when metaprograming, it's common to want to use `Object#respond_to_missing?`:

``` ruby
def respond_to_missing? name, include_private = false
  # Implementation details...
end
```

The use of `include_private = false`, in this case, is what causes this cop to flag the code as a violation. I agree, it is. ...but this is a standard library method which is out of my control as an implementer.

## Describe alternatives you've considered

I've been able temporarily work around this issue by using an exclude:

``` yaml
Style/OptionalBooleanParameter:
  Exclude:
    - "path/to/implementation"
```

This is not ideal as I'd rather not exclude the entire file but ignore a bad practice within the standard library that is out of my control instead.

**Comment by marcandre** [comment]
Good point.
We should definitely exclude `respond_to_missing?` by default. Can you think of any other builtin callback like that?
Adding the ability to exclude more methods would be a bonus (could be a separate PR)

**Comment by bkuhlmann** [comment]
> @marcandre Can you think of any other builtin callback like that?

Hmm, I don't know. `Object#respond_to_missing?` is what I see most often. I know there are others but when searching through my various code bases I came up empty for additional use cases. If I can find more, I'll post back here.

**Comment by natematykiewicz** [comment]
Can we start by fixing this for `respond_to_missing?`, and then fix it for more if more come up?

**Comment by bkuhlmann** [comment]
:bow: @natematykiewicz Yeah, I'd like that too. I've still not found another more pressing example other than `#respond_to_missing?` for the time being. This would be a good start, though!
